### PR TITLE
posters

### PR DESCRIPTION
**Pull request by blacktwin**
Initial work for #179. Adding poster objects first to more easily address issue #179. This PR can be a standlone as the rest of the request in #179 will be coming in a sperate PR.

### CODE REVIEW

File: plexapi/video.py [hunk 1]
```diff
@@ -89,6 +91,14 @@ def _defaultSyncTitle(self):
         """ Returns str, default title for a new syncItem. """
         return self.title
 
+    def _posters(self):
```
Comment (Hellowlol) [inline plexapi/video.py]: Just make this a normal function no _

File: plexapi/video.py [hunk 2]
```diff
@@ -42,6 +43,7 @@ def _loadData(self, data):
         self.type = data.attrib.get('type')
         self.updatedAt = utils.toDatetime(data.attrib.get('updatedAt'))
         self.viewCount = utils.cast(int, data.attrib.get('viewCount', 0))
+        self.posters = self._posters()
```
Comment (Hellowlol) [inline plexapi/video.py]: Delete this

File: plexapi/video.py [hunk 3]
```diff
@@ -24,6 +24,7 @@ class Video(PlexPartialObject):
             type (str): 'artist', 'album', or 'track'.
             updatedAt (datatime): Datetime this item was updated.
             viewCount (int): Count of times this item was accessed.
+            posters (List<:class:`~plexapi.media.Poster`>): List of poster objects.
```
Comment (Hellowlol) [inline plexapi/video.py]: Remove this.

File: tests/test_video.py [hunk 4]
```diff
@@ -101,6 +101,7 @@ def test_video_Movie_attrs(movies):
     assert [i.tag for i in movie.directors] == ['Nina Paley']
     assert movie.duration >= 160000
     assert movie.fields == []
```
Comment (Hellowlol) [inline tests/test_video.py]: Update the test.

**Comment by Hellowlol** [comment]
If it does a http call it should be a function or do the http when needed. With the current pr we will do a http call on every item plexapi builds.

**Comment by blacktwin** [comment]
So leave it as a function and not a attribute. 👍  Can do. Thanks!

File: plexapi/video.py [hunk 5]
```diff
@@ -89,6 +89,14 @@ def _defaultSyncTitle(self):
         """ Returns str, default title for a new syncItem. """
         return self.title
 
+    def posters(self):
+        """ Returns list of available poster objects. :class:`~plexapi.media.Poster`:"""
+        items = []
```
Comment (Hellowlol) [inline plexapi/video.py]: I think you can do ```return self.fetchItems('%s/posters', cls=media.Poster)```

Comment (blacktwin) [reply plexapi/video.py]: It doesn't appear like we can. The xml for posters does not have a `filter` or `tag` to be able to use the `fetchItems`. Edit above results in a empty list.

Comment (Hellowlol) [reply plexapi/video.py]: The for this item should be Photo, that the tag i get when i try to do this. Can change the tag and try again?

Comment (Hellowlol) [reply plexapi/video.py]: So it worked?

Comment (blacktwin) [reply plexapi/video.py]: Yup. Not 100% sure why, but it does and you seem to know how/why.

**Comment by Hellowlol** [comment]
Good stuff? Can you make a PR for art also?

**Pull request merged**